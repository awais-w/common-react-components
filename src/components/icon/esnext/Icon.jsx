import React from 'react';
import PropTypes from 'prop-types';
import bem from '@argos/utils/esnext/bem';
import './Icon.scss';

const Icon = (props) => {
  const { isCustom, icon, onClick, id, disabled, color } = props;
  const cn = bem({ prefix: 'cmc-', block: isCustom ? 'icon-custom' : 'icon' });
  return (
    <i
      className={`${cn(null, color, { disabled, [icon]: isCustom, clickable: onClick })}`}
      id={id || undefined}
      role={onClick ? 'button' : 'icon'}
      onClick={onClick}>
      {icon}
    </i>
  );
};

Icon.propTypes = {
  icon: PropTypes.string.isRequired,
  isCustom: PropTypes.bool,
  onClick: PropTypes.func,
  id: PropTypes.string,
  disabled: PropTypes.bool,
  color: PropTypes.oneOf([undefined, 'white', 'red', 'yellow', 'green', 'blue', 'black', 'grey']),
};

Icon.defaultProps = {
  onClick: null,
  isCustom: false,
};

export default Icon;
