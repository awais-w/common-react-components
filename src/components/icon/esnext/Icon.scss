@import '@cmc/common-styles/src/scss/_variables.scss';

@font-face {
  font-family: 'Material Icons';
  font-style: normal;
  font-weight: 400;
  src: local('Material Icons'), local('MaterialIcons-Regular'),
    url(https://fonts.gstatic.com/s/materialicons/v42/flUhRq6tzZclQEJ-Vdg-IuiaDsNcIhQ8tQ.woff2) format('woff2');
}

@font-face {
  font-family: 'icomoon';
  src: url('fonts/icomoon.woff?y5vhvj') format('woff');
  font-style: normal;
  font-weight: 400;
}

.cmc-icon {
  font-family: 'Material Icons';
  font-weight: normal;
  font-style: normal;
  font-size: $icon-size-standard;
  line-height: 1;
  letter-spacing: normal;
  text-transform: none;
  display: inline-block;
  white-space: nowrap;
  word-wrap: normal;
  direction: ltr;
  font-feature-settings: 'liga';
  -webkit-font-smoothing: subpixel-antialiased;
}

.cmc-icon-custom {
  /* use !important to prevent issues with browser extensions that change fonts */
  font-family: 'icomoon' !important;
  font-style: normal;
  font-weight: normal;
  font-variant: normal;
  text-transform: none;
  line-height: 1;
  font-size: 0 !important;

  /* Better Font Rendering =========== */
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  &:before {
    font-size: $font-size-large;
  }
}

i.cmc-icon,
i.cmc-icon-custom {
  &.clickable {
    cursor: pointer;
  }
  &--white {
    color: $color-white;
  }
  &--red {
    color: $color-red;
  }
  &--yellow {
    color: $color-star-stroke;
  }
  &--green {
    color: $color-green;
  }
  &--blue {
    color: $primary-color;
  }
  &--black {
    color: $color-black;
  }
  &--grey {
    color: $color-grey-dark;
  }
}

/* Custom icons */
.icon-change-adress:before {
  content: '\e900';
}
.icon-refund-delivery-charge:before {
  content: '\e914';
}
.icon-leader:before {
  content: '\e912';
}
.icon-url:before {
  content: '\e913';
}
.icon-currency:before {
  content: '\e90f';
}
.icon-to-left:before {
  content: '\e910';
}
.icon-to-right:before {
  content: '\e911';
}
.icon-address-lookup:before {
  content: '\e90e';
}
.icon-returns:before {
  content: '\e90d';
}
.icon-collected:before {
  content: '\e901';
}
.icon-created:before {
  content: '\e902';
}
.icon-delivered:before {
  content: '\e903';
}
.icon-in-progress:before {
  content: '\e904';
}
.icon-in-transit:before {
  content: '\e905';
}
.icon-ready-to-collect:before {
  content: '\e906';
}
.icon-refunded:before {
  content: '\e907';
}
.icon-return-failed:before {
  content: '\e908';
}
.icon-return-in-progress:before {
  content: '\e909';
}
.icon-returned:before {
  content: '\e90a';
}
.icon-rebook-delivery:before {
  content: '\e90b';
}
.icon-price-history:before {
  content: '\e90c';
}
.icon-pointer:before {
  content: '\e915';
}
.icon-contact-history:before {
  content: '\e916';
}
.icon-order-history:before {
  content: '\e917';
}
.icon-failed-payment:before {
  content: '\e918';
}
.icon-sticky-note:before {
  content: '\e919';
}
