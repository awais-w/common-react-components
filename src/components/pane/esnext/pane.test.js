import React from 'react';
import { shallow } from 'enzyme';
import Icon from '@cmc/icon/esnext/Icon';
import Skeleton from 'react-loading-skeleton';
import Pane from './pane';

const defaultProps = {
  title1: 'Title 1',
  title2: 'Title 2',
  title3: 'Title 3',
  title4: 'Title 4',
  icon: { name: 'local_shipping', isCustom: false },
  isLoading: false,
  collapsed: false,
};

const setup = ({ props }) => {
  const component = shallow(<Pane {...props} />);

  return {
    wrapper: component,
    getPane: () => component.find('.ac-pane'),
    getHeading: () => component.find('.ac-pane__header'),
    getTitleWrapper: () => component.find('.ac-pane__title'),
    getTitle1: () => component.find('.ac-pane__title1'),
    getTitle2: () => component.find('.ac-pane__title2'),
    getTitle3: () => component.find('.ac-pane__title3'),
    getTitle4: () => component.find('.ac-pane__title4'),
    getTitleRight: () => component.find('.ac-pane__title-right'),
    getIcon: () => component.find(Icon),
    getBody: () => component.find('.ac-pane__body'),
    getChildren: () => component.find('.child'),
    getLoading: () => component.find(Skeleton),
    getCollapsed: () => component.find('.ac-pane__body--collapsed'),
  };
};

describe('<Pane />', () => {
  describe('with default props', () => {
    const component = setup({ props: defaultProps });

    it('should match default snapshot', () => {
      expect(component.wrapper).toMatchSnapshot();
    });

    it('should render a <Pane /> wrapper', () => {
      expect(component.getPane().exists()).toBe(true);
    });

    it('should render a heading', () => {
      expect(component.getHeading().exists()).toBe(true);
    });

    it('should render a <H3 /> title wrapper', () => {
      expect(component.getTitleWrapper().exists()).toBe(true);
      expect(component.getTitleWrapper().type()).toEqual('h3');
    });

    it('should render title1', () => {
      expect(component.getTitle1().exists()).toBe(true);
    });

    it('should render title2', () => {
      expect(component.getTitle2().exists()).toBe(true);
    });
  });

  describe('with custom props', () => {
    describe('with a title3 prop', () => {
      const title3 = 'wumbo';
      const component = setup({ props: { ...defaultProps, title3 } });

      it('should render a title3', () => {
        expect(component.getTitle3().exists()).toBe(true);
      });
    });

    describe('with a title4 prop', () => {
      const title4 = 'gumbo';
      const component = setup({ props: { ...defaultProps, title4 } });

      it('should render a title4', () => {
        expect(component.getTitle4().exists()).toBe(true);
      });
    });

    describe('with icon prop', () => {
      const component = setup({ props: { ...defaultProps } });

      it('should render an icon', () => {
        expect(component.getIcon().exists()).toBe(true);
      });
    });

    describe('with a titleRight prop', () => {
      const titleRight = 'wumbo';
      const component = setup({ props: { ...defaultProps, titleRight } });

      it('should render a titleRight', () => {
        expect(component.getTitleRight().exists()).toBe(true);
      });
    });

    describe('collapsed', () => {
      const component = setup({ props: { ...defaultProps, collapsed: true } });

      it('should be collapsed if collapsed is true', () => {
        expect(component.getCollapsed().exists()).toBeTruthy();
      });

      it('should have body not clickable if its not collapsed', () => {
        component.wrapper.setProps({ ...defaultProps, collapsed: false });
        expect(typeof component.getBody().props().onClick === 'function').toBeFalsy();
      });

      it('should toggle on header click', () => {
        component.getHeading().props().onClick();
        expect(component.getCollapsed().exists()).toBeTruthy();
      });

      it('should toggle on body click when its in collapsed state', () => {
        component.getBody().props().onClick();
        expect(component.getCollapsed().exists()).toBeFalsy();
      });
    });

    describe('with children', () => {
      const children = [
        <div key='1' className='child'>
          Child 1
        </div>,
        <div key='2' className='child'>
          Child 2
        </div>,
        <div key='3' className='child'>
          Child 3
        </div>,
      ];

      it('should render skeleton loading if isLoading is true', () => {
        const component = setup({ props: { ...defaultProps, children, isLoading: true } });
        expect(component.getChildren().length).toEqual(0);
        expect(component.getLoading().exists()).toBeTruthy();
      });
      it('should render children props', () => {
        const component = setup({ props: { ...defaultProps, children } });
        expect(component.getBody().exists()).toBe(true);
        expect(component.getChildren().length).toEqual(3);
      });
    });
  });
});
