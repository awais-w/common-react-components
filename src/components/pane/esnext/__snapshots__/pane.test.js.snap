// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pane /> with default props should match default snapshot 1`] = `
<section
  className="ac-pane ac-pane--wrapper"
>
  <div
    className="ac-pane__header"
    onClick={[Function]}
  >
    <div
      className="ac-pane__header-wrapper"
    >
      <Icon
        icon="local_shipping"
        isCustom={false}
        onClick={null}
      />
      <h3
        className="ac-pane__title h3"
      >
        <span
          className="ac-pane__title1 title-item"
        >
          Title 1
        </span>
        <span
          className="ac-pane__title2 title-item"
        >
          Title 2
        </span>
        <span
          className="ac-pane__title3 title-item"
        >
          Title 3
        </span>
        <span
          className="ac-pane__title4 title-item"
        >
          Title 4
        </span>
      </h3>
    </div>
  </div>
  <div
    className="ac-pane__body ac-pane__body--false ac-pane__body--false"
    onClick={null}
  />
</section>
`;
